schemas/collector-report.schema.json

```
@@ -0,0 +1,75 @@
+{
+  "$schema": "http://json-schema.org/schema",
+
+  "definitions": {
+    "exposure": {
+      "type": "string",
+      "enum": ["ServiceWorker", "SharedWorker", "Window", "Worker"]
+    },
+
+    "feature_name": {
+      "type": "string",
+      "pattern": "^[a-zA-Z_0-9-$@.]*$",
+      "description": "Dot separated identifier path.\n\n@example \"javascript.builtins.Array.at\"",
+      "examples": ["javascript.builtins.Array.at"]
+    },
+
+    "collector_result": {
+      "oneOf": [
+        {
+          "type": "object",
+          "properties": {
+            "exposure": { "$ref": "#/definitions/exposure" },
+            "name": { "$ref": "#/definitions/feature_name" },
+            "result": { "type": "boolean" }
+          },
+          "additionalProperties": false,
+          "required": ["exposure", "name", "result"]
+        },
+        {
+          "type": "object",
+          "properties": {
+            "exposure": { "$ref": "#/definitions/exposure" },
+            "message": {
+              "type": "string",
+              "pattern": "^threw .*$|^Testing .* in workers is not yet implemented$|^Browser does not support detection methods$",
+              "description": "Reason a result could not be collected.\n\n@example \"threw TypeError: Failed to construct 'Notification': Illegal constructor.\"\n@example \"Testing CanvasRenderingContext2D in workers is not yet implemented\"\n@example \"Browser does not support detection methods\""
+            },
+            "name": { "$ref": "#/definitions/feature_name" },
+            "result": { "type": "null" }
+          },
+          "additionalProperties": false,
+          "required": ["exposure", "message", "name", "result"]
+        }
+      ]
+    }
+  },
+
+  "title": "CollectorReport",
+  "type": "object",
+  "properties": {
+    "__version": {
+      "type": "string",
+      "pattern": "^[0-9.]+$",
+      "description": "Version of mdn-bcd-collector that collected the report."
+    },
```

The new schema, since v8.2.0 of the collector, also contains data on browser extensions that may conflict with collected results:

```suggestion
    },
    "extensions": {
      "type": "object",
      "patternProperties": {
        "^[a-zA-Z_0-9]*$": {
          "type": "string",
          "enum": ["chrome", "firefox", "safari"]
        }
      }
    }
```

---

Is `extensions` required?

-----

Is extensions a list of installed browser extensions like for Chrome those installed from the Chrome Web Store? I'd like to include information on it in the markdown file.

---

Is there an example of `extensions` I can look at. I didn't see any in main in https://github.com/GooborgStudios/mdn-bcd-results.

---

Let's treat this as a follow up. You've already checked the schema against the files in https://github.com/GooborgStudios/mdn-bcd-results, so if we have a report with `extensions` we can update the schema.

---

I've filed https://github.com/mdn/browser-compat-data/issues/20102.